drv/transceivers-api/src/lib.rs

```diff
@@ -25,16 +27,60 @@ impl From<FpgaError> for TransceiversError {
 
 /// Each field is a bitmask of the 32 transceivers in big endian order, which
 /// results in Port 31 being bit 31, and so forth.
-#[derive(Copy, Clone, zerocopy::FromBytes, zerocopy::AsBytes)]
+#[derive(Copy, Clone, FromBytes, AsBytes)]
 #[repr(C)]
 pub struct ModulesStatus {
     pub enable: u32,
     pub reset: u32,
     pub lpmode_txdis: u32,
     pub power_good: u32,
-    pub power_good_timeout: u32,
     pub present: u32,
     pub irq_rxlos: u32,
+    pub power_good_timeout: u32,
+    pub power_good_fault: u32,
+}
+
+/// The power states we use to model transceiver state
+/// A4 - Module not present.
+/// A3 - Module present and powered off.
+/// A2 - Module powered, out of reset, and in low-power mode
+/// A0 - Module is in high-power mode.
+/// Fault - A power fault has ocurred and must be cleared.
+#[derive(
+    Copy, Clone, PartialEq, Eq, FromPrimitive, AsBytes, Serialize, Deserialize,
+)]
+#[repr(u8)]
+pub enum PowerState {
+    A4 = 0,
+    A3 = 1,
+    A2 = 2,
+    A0 = 3,
+    Fault = 4,
+}
+
+impl From<u8> for PowerState {
+    fn from(v: u8) -> PowerState {
+        match v {
+            0 => PowerState::A4,
+            1 => PowerState::A3,
+            2 => PowerState::A2,
+            3 => PowerState::A0,
+            4 => PowerState::Fault,
+            _ => unreachable!(),
```

Comment:
I'm skeptical about non-total implementations of `from`, since they lead to Surprise Crashes.

The cleanest API design would be to make this `TryFrom`, e.g.
```rust
impl TryFrom<u8> for PowerState {
    type Error = ();
    fn try_from(v: u8) -> Result<PowerState, Self::Error> {
        match v {
            0 => Ok(PowerState::A4),
            1 => Ok(PowerState::A3),
            2 => Ok(PowerState::A2),
            3 => Ok(PowerState::A0),
            4 => Ok(PowerState::Fault),
            _ => Err(()),
        }
    }
}
```

(you'd then need to add an `unwrap` to `all_power_states`, which is the only place that uses this conversion)